app/app/api/api_v1/endpoints/terms.py

```diff
@@ -56,52 +57,30 @@ def add_term_students_by_id(
     _: models.Admin = Depends(deps.get_current_admin_with_permission("term")),
 ) -> Any:
     if term := crud.term.get(db, id=term_id):
-        errors = []
+        response: Dict[str, Any] = {"success": []}
+        errors = defaultdict(lambda: [])
 
         for user_id in user_ids:
             if user := crud.user.get(db, id=user_id):
                 if user.type == "student":
                     if user.school_id == term.year.school_id:
                         if student := crud.student.get(db, id=user_id):
                             crud.student.update(db, db_obj=student, obj_in=StudentUpdate(term_id=term_id))
+                            response["success"].append(student.user_id)
                         else:
-                            errors.append(
-                                {
-                                    "msg": "no student object",
-                                    "type": "user.no_student_object",
-                                    "loc": ["user ID", user_id],
-                                }
-                            )
+                            errors["no student object"].append(user_id)
                     else:
-                        errors.append(
-                            {
-                                "msg": "different schools",
-                                "type": "objects.different_school",
-                                "loc": ["user ID", user_id],
-                            }
-                        )
+                        errors["different schools"].append(user_id)
                 else:
-                    errors.append(
-                        {
-                            "msg": "not a student",
-                            "type": "user.not_a_student",
-                            "loc": ["user ID", user_id],
-                        }
-                    )
+                    errors["not a student"].append(user_id)
             else:
-                errors.append(
-                    {
-                        "msg": "not a user",
-                        "type": "user.not_a_user",
-                        "loc": ["user ID", user_id],
-                    }
-                )
-
-        if errors:
-            raise HTTPException(status_code=418, detail=errors)
-
-        db.refresh(term)
-        return term.students
+                errors["not a user"].append(user_id)
+
+        # Cleanup response a bit
+        if len(response["success"]) == 0:
+            del response["success"]
+        response["errors"] = errors
+        return response
```

Comment:
How about we just make response a defaultdict, append IDs to `response["success"]` and `response["errors"]["msg"]` and just specify `Dict[str, Union[List[int], Dict[str, List[int]]]]` as the response model?

Reply:
k nvm, I get why it won't work